src/server/controllers/renderer/renderer.tsx

```diff
@@ -46,34 +54,52 @@ export async function get(req: Request, res: Response) {
       </Provider>
     </ApolloProvider>
   );
+  runSaga();
 
-  try {
-    const extractor = new ChunkExtractor({ statsFile });
-    const tree = extractor.collectChunks(<App />);
-
-    await Promise.all([
-      // kick apollo, redux-saga, and styled-components
-      getMarkupFromTree({
-        tree,
-        renderFunction: renderToStaticMarkup
-      }),
-      runSaga()
-    ]);
+  const preloadedState = JSON.stringify(store.getState());
+  const initialState = escape(preloadedState);
+  const graphql = JSON.stringify(client.extract());
 
-    const body = renderToString(tree);
-    const preloadedState = JSON.stringify(store.getState());
-    const helmetContent = Helmet.renderStatic();
-    const meta = `
+  const extractor = new ChunkExtractor({ statsFile });
+  const tree = extractor.collectChunks(<App />);
+  const helmetContent = Helmet.renderStatic();
+  const meta = `
       ${helmetContent.meta.toString()}
       ${helmetContent.title.toString()}
     `.trim();
-    const style = sheet.getStyleTags();
-    const scripts = extractor.getScriptTags({ nonce });
-    const graphql = JSON.stringify(client.extract());
+  await Promise.all([
+    // kick apollo, redux-saga, and styled-components
+    getMarkupFromTree({
+      tree,
+      renderFunction: renderToStaticMarkup
+    }),
+    runSaga()
+  ]);
+
+  res.write(
+    `<!DOCTYPE html>
```

Comment:
Please delete them because we moved them to renderFullPage.ts.

Reply:
actually it works like that and renderFullPage is not needed